### ISSUE TITLE
Different results when converting yolov5 7.0 model

### ISSUE THREAD

**Issue by ZiyueWangUoB** [Tencent/ncnn#4784]
Following the guide from #4541, I converted model from pytorch -> torchscript -> NCNN. Inference can run but the result is wrong, I end up with different results. The two results are similar but not the same, which causes a bit drop in accuracy.

Output from NCNN
`[[6.84316444e+00 6.09831715e+00 1.48272419e+01 ... 4.08707978e-03
  5.32282575e-04 6.46748289e-04]
 [1.26748819e+01 6.36044884e+00 2.62327232e+01 ... 2.93191546e-03
  5.02826064e-04 5.72151330e-04]
 [1.76190014e+01 6.70258999e+00 3.29912796e+01 ... 2.49496964e-03
  7.09522341e-04 8.31600686e-04]`

Output from ONNX
`[[6.8614807e+00 5.4661732e+00 1.4046919e+01 ... 2.0090640e-03
  1.0010898e-03 1.0457277e-02]
 [1.2899274e+01 6.0367393e+00 2.5906830e+01 ... 1.6336143e-03
  9.8460913e-04 1.0029525e-02]
 [1.8008467e+01 6.2882919e+00 3.0417656e+01 ... 2.1933317e-03
  1.0814071e-03 2.2881538e-02]`

Output from Torchscript
`(tensor([[[6.8615e+00, 5.4662e+00, 1.4047e+01,  ..., 2.0091e-03,
          1.0011e-03, 1.0457e-02],
         [1.2899e+01, 6.0367e+00, 2.5907e+01,  ..., 1.6336e-03,
          9.8461e-04, 1.0030e-02],
         [1.8008e+01, 6.2883e+00, 3.0418e+01,  ..., 2.1933e-03,
          1.0814e-03, 2.2881e-02]`

**Comment by wzyforgit** [comment]
是最新版的ncnn嘛？因为之前出过vulkan推理错误的bug，在后续版本修复了
参见issue：https://github.com/Tencent/ncnn/issues/4685

**Comment by ZiyueWangUoB** [comment]
@wzyforgit 是的，我现在用了最新的，没问题了。 谢谢

**Comment by glenn-jocher** [comment]
@ZiyueWangUoB @wzyforgit I have great news 😃! I've recently added official support for Ultralytics YOLOv8 NCNN export ✅ in PR https://github.com/ultralytics/ultralytics/pull/3529 with the help of @nihui which is part of `ultralytics==8.0.129`. NCNN works for all tasks including Detect, Segment, Pose and Classify.


You can now export with CLI:
```
yolo export model=yolov8n.pt format=ncnn
```

or Python:
```python
from ultralytics import YOLO

# Create a model
model = YOLO('yolov8n.pt')

# Export the model to NCNN with arguments
model.export(format='ncnn', half=True, imgsz=640)
```

Output is a `yolov8n_ncnn_model/` directory containing `model.bin`, `model.param` and `metadata.yaml`, along with extra PNNX files. For details see https://github.com/pnnx/pnnx README.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Please let us know if NCNN export is working correctly for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
